ios/Classes/PTPluginUtils.m

```diff
@@ -20,26 +58,385 @@ + (NSNumber *)PT_idAsNSNumber:(id)value
+    }
+    
+    flutterResult(nil);
+}
+
++ (void)exportAnnotations:(NSString *)annotationList resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+        
+        if (!annotationList) {
+            PTFDFDoc *fdfDoc = [doc FDFExtract:e_ptboth];
+            flutterResult([fdfDoc SaveAsXFDFToString]);
+        } else {
+            PTVectorAnnot *annots = [[PTVectorAnnot alloc] init];
+            
+            
+            NSArray *array = [PTPluginUtils PT_idAsArray:[PTPluginUtils PT_JSONStringToId:annotationList]];
+            
+            for (NSDictionary *annotation in array) {
+                NSString *annotationId = annotation[PTAnnotIdKey];
+                int pageNumber = ((NSNumber *)annotation[PTAnnotPageNumberKey]).intValue;
+                if (annotationId.length > 0) {
+                    PTAnnot *annot = [PTPluginUtils findAnnotWithUniqueID:annotationId onPageNumber:pageNumber documentViewController:docVC];
+                    if ([annot IsValid]) {
+                        [annots add:annot];
+                    }
+                }
+            }
+            
+            if ([annots size] > 0) {
+                PTFDFDoc *fdfDoc = [doc FDFExtractAnnots:annots];
+                flutterResult([fdfDoc SaveAsXFDFToString]);
+            } else {
+                flutterResult(nil);
+            }
+        }
+        
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to export annotations. %@", error.localizedDescription);
+    }
+}
+
+
++ (void)flattenAnnotations:(bool)formsOnly resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    [docVC.toolManager changeTool:[PTPanTool class]];
+    
+    NSError *error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+        
+        [doc FlattenAnnotations:formsOnly];
+        
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to flatten annotations. %@", error.localizedDescription);
+    }
+    
+    flutterResult(nil);
```

Comment:
As before, it's odd to me we don't call this on error. Is there no way to pass back a success or failure token?